File: answers/leetcode/349. Intersection of Two Arrays/Solution.py
class Solution:
    def intersection(self, nums1: List[int], nums2: List[int]) -> List[int]:
        number_set: Set[int] = set()
        for number in nums1:
            number_set.add(number)
        result: Set[int] = set()
        for number in nums2:
            if number in number_set:
                result.add(number)
        return list(result)

File: answers/leetcode/349. Intersection of Two Arrays/Solution.java
class Solution {
    public int[] intersection(int[] nums1, int[] nums2) {
        Set<Integer> nums1Set = new HashSet<Integer>();
        for (int num : nums1) {
            nums1Set.add(num);
        }
        Set<Integer> intersect = new HashSet<Integer>();
        for (int num : nums2) {
            if (nums1Set.contains(num)) {
                intersect.add(num);
            }
        }
        int[] result = new int[intersect.size()];
        int i = 0;
        for (int num : intersect) {
            result[i++] = num;
        }
        return result;
    }
}